modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2538,7 +2538,7 @@ def sort_function(df):  # pragma: no cover
         # We perform the final steps of the sort on full axis partitions, so we know that the
         # length of each partition is the full length of the dataframe.
         if self.has_materialized_columns:
-            self._set_axis_lengths_cache([len(self.columns)], axis=axis.value ^ 1)
+            result._set_axis_lengths_cache([len(self.columns)], axis=axis.value ^ 1)
```

Comment:
This looks more like a typo, I don’t think it’s necessary to add a test from the issue.